### PR TITLE
Fix fetching boolean data as binary

### PR DESCRIPTION
**Pull request by axelboc**
This follows https://github.com/silx-kit/h5web/pull/1692 and https://github.com/usnistgov/h5wasm/pull/76

### CODE REVIEW

File: h5grove/utils.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def _sanitize_dtype(dtype: np.dtype) -> np.dtype:
 
     :raises ValueError: If trying to sanitize a non-numeric numpy dtype
     """
-    if dtype.kind not in ("f", "i", "u"):
```
Comment (loichuder) [inline h5grove/utils.py:215]: Add a little comment that explains why `b` is actually numeric ?

**Review by loichuder — APPROVED** [review]

File: h5grove/utils.py (hunk 1 above)
Comment (axelboc) [reply h5grove/utils.py:215]: Well, `b` means `byte`, not boolean, so it was more of an oversight.

Now that you mention it, I feel like this check doesn't really make sense. The purpose of `_sanitize_dtype` is to make specific dtypes safe:

- big endian to little endian
- float16 to float32
- float128 to float64

It shouldn't care about anything else. The `convert` function, however, does check that `dtype=safe` is only used with numeric datasets as you would expect. It does so by calling `is_numeric_data`, which does check for booleans already.

I'll push a separate commit with a proposed refactoring.

File: h5grove/utils.py [hunk 2]
```diff
@@ -208,25 +208,21 @@ def get_type_metadata(type_id: h5py.h5t.TypeID) -> TypeMetadata:
...
         return np.dtype("<f4")
 
     # Convert float128 to float64 (unavoidable loss of precision)
-    if result.kind == "f" and result.itemsize > 8:
+    if dtype.kind == "f" and dtype.itemsize > 8:
         return np.dtype("<f8")
 
-    return result
+    # Convert big-endian to little-endian
+    if dtype.byteorder == '>':
```
[the remark this answers is not in the document]
Comment (loichuder) [reply h5grove/utils.py]: Probably ? :shrug:

**Review by loichuder — APPROVED** [review]

File: h5grove/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (t20100) [reply h5grove/utils.py]: Native can be either big-endian or little-endian so best to convert.

File: h5grove/utils.py [hunk 3]
```diff
@@ -208,25 +208,21 @@ def get_type_metadata(type_id: h5py.h5t.TypeID) -> TypeMetadata:
...
-    if result.kind == "f" and result.itemsize > 8:
+    if dtype.kind == "f" and dtype.itemsize > 8:
         return np.dtype("<f8")
 
-    return result
+    # Convert native and big-endian byte-orders to little-endian
+    if dtype.byteorder == "=" or dtype.byteorder == ">":
+        return dtype.newbyteorder("<")
+
+    return dtype
```
Comment (t20100) [inline h5grove/utils.py:225]: This would also does the job:
```suggestion
    return dtype.newbyteorder("<")
```

Comment (axelboc) [reply h5grove/utils.py:225]: Does this work on any dtype, even non-numeric? It feels wrong to do this for, say `|O` or `|S5`.

Comment (t20100) [reply h5grove/utils.py:225]: Yes indeed, even if it looks to works for `‘|’ not applicable` endianness, it is not a good idea to do so IMO.
Forget my comment ;)

**Pull request merged**